src/pyenphase/envoy.py

```diff
@@ -373,39 +374,39 @@ async def _request(
                 method if method else "POST",
                 url,
                 headers={**DEFAULT_HEADERS, **self.auth.headers},
-                follow_redirects=True,
-                auth=self.auth.auth,
                 timeout=self._timeout,
-                content=orjson.dumps(data),
+                data=orjson.dumps(data),
+                middlewares=middlewares,
             )
         else:
             _LOGGER.debug("Requesting %s with timeout %s", url, self._timeout)
             response = await self._client.get(
                 url,
                 headers={**DEFAULT_HEADERS, **self.auth.headers},
-                follow_redirects=True,
-                auth=self.auth.auth,
                 timeout=self._timeout,
+                middlewares=middlewares,
             )
 
-        status_code = response.status_code
+        status_code = response.status
         if status_code in (HTTPStatus.UNAUTHORIZED, HTTPStatus.FORBIDDEN):
             raise EnvoyAuthenticationRequired(
                 f"Authentication failed for {url} with status {status_code}, "
                 "please check your username/password or token."
             )
+
+        # Read the content to cache it
+        content = await response.read()
         # show all responses centrally when in debug
         if debugon:
             request_end = time.monotonic()
             content_type = response.headers.get("content-type")
-            content = response.content
             _LOGGER.debug(
                 "Request reply in %s sec from %s status %s: %s %s",
                 round(request_end - request_start, 1),
                 url,
                 status_code,
                 content_type,
-                content,
+                content,  # Use the actual content bytes
             )
```

Comment:
Only read response when in debug. Aiohttp will cache the response.

```suggestion
        # show all responses centrally when in debug
        if debugon:
            request_end = time.monotonic()
            content_type = response.headers.get("content-type")
            _LOGGER.debug(
                "Request reply in %s sec from %s status %s: %s %s",
                round(request_end - request_start, 1),
                url,
                status_code,
                content_type,
                await response.read()
            )
```

Reply:
f403e9d